extensions/podman/packages/extension/src/extension.spec.ts

```diff
@@ -2501,6 +2501,37 @@ describe('registerOnboardingRemoveUnsupportedMachinesCommand', () => {
   });
 });
 
+describe('socketCompatibilityMode command', () => {
+  test('check telemetry sent when error on check disguised podman Mac', async () => {
+    vi.mocked(extensionApi.env).isMac = true;
+    vi.mocked(extensionApi.commands.registerCommand).mockReturnValue({ dispose: vi.fn() });
+
+    await extension.activate(getContextMock());
+
+    const socketCommand = 'podman.socketCompatibilityMode';
+
+    // check command is called
+    expect(extensionApi.commands.registerCommand).toHaveBeenCalledWith(socketCommand, expect.any(Function));
+
+    const func = vi
+      .mocked(extensionApi.commands.registerCommand)
+      .mock.calls.find(([command]) => command === socketCommand)?.[1];
+
+    assert(func, `${socketCommand} should have been registered`);
+
+    const error = new Error('test error');
+
+    vi.mocked(isDisguisedPodman).mockRejectedValue(error);
+    // call the function
+    await func();
+
+    expect(isDisguisedPodman).toBeCalled();
```

Comment:
Please add `expect(isDisguisedPodman).not.toBeCalled();` before the `await func();`